library/src/interactor_impl.cxx

```diff
@@ -71,8 +75,102 @@ class interactor_impl::internals
 #endif
   }
 
+  //----------------------------------------------------------------------------
+  // Method defined to normalize the Z axis so all models are treated temporarily
+  // as Z-up axis models.
+  void zUpTransforms(vtkTransform* to, vtkTransform* from)
+  {
+    vtkRenderer* renderer =
+      this->VTKInteractor->GetRenderWindow()->GetRenderers()->GetFirstRenderer();
+    const double* up = renderer->GetEnvironmentUp();
+    const double* right = renderer->GetEnvironmentRight();
+    double fwd[3];
+    vtkMath::Cross(right, up, fwd);
+    const double m[16] = {
+      right[0], right[1], right[2], 0, //
+      fwd[0], fwd[1], fwd[2], 0,       //
+      up[0], up[1], up[2], 0,          //
+      0, 0, 0, 1,                      //
+    };
+    to->SetMatrix(m);
```

Comment:
Set all elements using `to->SetElement(0, 0, right[0])` (9 calls in total)

Reply:
```cpp
    const double m[9] = {
      right[0], right[1], right[2], //
      fwd[0], fwd[1], fwd[2],       //
      up[0], up[1], up[2],          //
    };
    to->DeepCopy(m);
```